runtime/altair/src/governance/mod.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! OpenGov Pallet implementations
+
+use cfg_primitives::{TechnicalCollective, TechnicalMembership};
+use frame_support::{parameter_types, traits::EitherOf};
+use frame_system::EnsureRootWithSuccess;
+pub use runtime_common::origins::gov::{
+	pallet_custom_origins, ReferendumCanceller, ReferendumKiller, WhitelistedCaller,
+};
+
+use super::*;
+mod tracks;
+pub use tracks::TracksInfo;
+
+impl pallet_conviction_voting::Config for Runtime {
+	type Currency = Balances;
+	type MaxTurnout =
+		frame_support::traits::tokens::currency::ActiveIssuanceOf<Balances, Self::AccountId>;
+	type MaxVotes = ConstU32<512>;
+	type Polls = Referenda;
+	type RuntimeEvent = RuntimeEvent;
+	type VoteLockingPeriod = EnactmentPeriod;
+	type WeightInfo = weights::pallet_conviction_voting::WeightInfo<Self>;
+}
+
+parameter_types! {
+	pub const AlarmInterval: BlockNumber = 1;
+	pub const SubmissionDeposit: Balance = 10 * AIR;
+	pub const UndecidingTimeout: BlockNumber = 7 * DAYS;
+}
+
+parameter_types! {
+	pub const MaxBalance: Balance = Balance::max_value();
+}
+pub type TreasurySpender = EnsureRootWithSuccess<AccountId, MaxBalance>;
+
+impl pallet_custom_origins::Config for Runtime {}
+
+impl pallet_whitelist::Config for Runtime {
```

Comment:
NIT and open to discussion:

I feel that splitting the runtime config implementations into different files ends up reducing the readability and maintainability because both splitting files end up with mutual self-references using types from the other, and sometimes it is very difficult to find the implementation for a pallet if these files grow

I'm a big fan of splitting things, but I would leave all `impl xyz::Config for Runtime` parts in the same file as an exception.

Reply:
I already knew you would complain about this. Absolutely fine for me to move it into the same place. I just preferred having all separate as long as this PR is a draft and not open for reviews :)

Reply:
Sorry for my early review! 🙏🏻 Just wanna to highlight the topic